File: examples/python/database/mysql.py
from __future__ import annotations

import asyncio
import os
from urllib.parse import urlparse

from identifold.storage import ReferenceReservation
from identifold.storage.mysql import MySQLStorageAdapter


async def main() -> None:
    from mysql.connector.aio import connect

    value = os.environ["IDENTIFOLD_TEST_MYSQL_URL"]
    url = urlparse(value)
    connection = await connect(
        host=url.hostname or "127.0.0.1",
        port=url.port or 3306,
        user=url.username,
        password=url.password,
        database=url.path.lstrip("/"),
        autocommit=True,
    )
    try:
        adapter = MySQLStorageAdapter(connection)
        request = ReferenceReservation(
            "01890f8c-7b2a-7cc3-98b0-112233445568",
            "order",
            "ORD-9876-5432-10-X",
        )
        reserved = await adapter.reserve(request)
        mapping = await adapter.resolve(request.reference, request.namespace)
        print({"reserved": reserved, "mapping": mapping})
    finally:
        await connection.close()


if __name__ == "__main__":
    asyncio.run(main(), loop_factory=asyncio.SelectorEventLoop)

File: examples/ruby/database/mysql.rb
require "identifold"
require "identifold/storage/mysql"
require "mysql2"
require "uri"

url = URI(ENV.fetch("IDENTIFOLD_TEST_MYSQL_URL"))
connection = Mysql2::Client.new(
  host: url.host,
  port: url.port || 3306,
  username: url.user,
  password: url.password,
  database: url.path.delete_prefix("/"),
  cast_booleans: true
)
adapter = Identifold::Storage::MySQLAdapter.new(connection)
request = Identifold::Storage::ReferenceReservation.new(
  machine_id: "01890f8c-7b2a-7cc3-98b0-112233445568",
  namespace: "order",
  reference: "ORD-9876-5432-10-X"
)
reserved = adapter.reserve(request)
mapping = adapter.resolve(request.reference, request.namespace)
puts({ reserved: reserved, mapping: mapping }.inspect)
connection.close
